addon/scheduler.js

```diff
@@ -0,0 +1,68 @@
+import RSVP from 'rsvp';
+import { run } from '@ember/runloop';
+
+const USE_RAF = typeof requestAnimationFrame === 'function';
+let _didTransition;
+let _afterFirstRoutePaint;
+let _afterContentPaint;
+
+reset();
+
+export function beginTransition() {
+  _didTransition = RSVP.defer();
+  _afterFirstRoutePaint = _didTransition.promise.then(() => afterNextPaint());
+  _afterContentPaint = _afterFirstRoutePaint.then(() => afterNextPaint());
```

Comment:
I suspect this queue is meaningless and not what we want and we can simply drop it from this re-imagined implementation.

Alternatively, we can fix it. It seems folks want the ability to schedule "do some work after the next paint" from an already active and rendered route. That is not what this queue accomplishes.

Reply:
I want to change it to use requestIdleCallback if available

Reply:
@rwjblue seems orthogonal to the question of whether this queue is even meaningful. At the moment it is implemented as `afterSecondRoutePaint` which is not what I think we want.

Reply:
Yes, this is actually `do something after second paint`. What do you think we want here?